### ISSUE TITLE
Bug: no-side-effects-in-computed-properties on no-side-effect code

### ISSUE THREAD

**Issue by Hacktivista** [vuejs/eslint-plugin-vue#511]
**Tell us about your environment**
"eslint": "^4.6.1"
"eslint-plugin-vue": "^4.4.0"

* **ESLint Version:** v4.19.1
* **eslint-plugin-vue Version:** 4.6.1
* **Node Version:** v6.11.2

**Please show your full configuration:**

```
{
  "extends": [
    "airbnb-base",
    "plugin:vue/recommended",
  ],
  rules: {
    "vue/max-attributes-per-line": "warn",
    "vue/html-self-closing": "warn",
  },
  "env": {
    "browser": true,
    "jquery": true,
  },
  globals: {
    "Auth": true,
    "axios": true,
    "Vue": true,
    "trans": true,
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

```vue
    permissionsOutput() {
      const permissions = [];
      this.subjectType.forEach(subject => (
        permissions.push(subject.x)
      ));
      return permissions;
    }
```

**What did you expect to happen?**
Should not throw errors, since I'm creating the `const permissions` inside the computer property.

**What actually happened? Please include the actual, raw output from ESLint.**
```
  152:7   error    Unexpected side effect in "permissionsOutput" computed property  vue/no-side-effects-in-computed-properties
```

**Comment by michalsnik** [comment]
Please check `v4.7.0` I was fixing this rule recently and seems to be working fine for the given case now ;)
